{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from samap.mapping import SAMAP\n",
    "from samap.analysis import (get_mapping_scores, GenePairFinder, transfer_annotations,\n",
    "                             sankey_plot, chord_plot, CellTypeTriangles, \n",
    "                             ParalogSubstitutions, FunctionalEnrichment,\n",
    "                             convert_eggnog_to_homologs, GeneTriangles)\n",
    "from samalg import SAM\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Tutorial modified from https://github.com/atarashansky/SAMap/blob/main/SAMap_vignette.ipynb"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Running BLAST"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "SAMap requires BLAST results between all pairs of species; in this case, the salamander Pleurodeles waltl (`pw`), the turtle Trachemys scripta / Chrysemys picta (`cp`), and the lizard Pogona vitticeps (`pv`).\n",
    "\n",
    "BLAST was run with a custom bash script `map_genes.sh` written by A. Tarashansky https://github.com/atarashansky/SAMap; the script auto-generates a directory called `maps/` in which the results will be deposited. The precomputed BLAST outputs are provided in the folder `SAMap_maps/`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Reciprocal BLAST hits will be used to construct the gene homology graph used by SAMap."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Running SAMap"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "SAMap accepts file paths to unprocessed, raw `.h5ad` files; these files were obtained from Seurat objects using the convertFormat function from the R package sceasy (from=\"Seurat\", to=\"anndata\").  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading in raw data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "fn3 = '/burg/tosches/users/mt3353/SAMap/datasets/Cp_neurons_3sp_integration_final.h5ad'\n",
    "fn1 = '/burg/tosches/users/mt3353/SAMap/datasets/Pw_neurons_3sp_integration_final.h5ad'\n",
    "fn2 = '/burg/tosches/users/mt3353/SAMap/datasets/Pv_neurons_3sp_integration_final.h5ad'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "SAMap expects the above to be in a dictionary keyed by the species IDs determined in the BLAST step:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "filenames = {'cp': fn3, 'pw':fn1,'pv':fn2}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "SAMAP objects instantiated below. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sm = SAMAP(\n",
    "        filenames,\n",
    "        f_maps = '/burg/tosches/users/mt3353/SAMap/processed/',\n",
    "        save_processed=True #if False, do not save the processed results to `*_pr.h5ad`\n",
    "   )"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "## Loading in SAM objects directly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Start here if the SAM objects have been created and saved already\n",
    "\n",
    "fn3 = '/burg/tosches/users/mt3353/SAMap/datasets/Cp_neurons_3sp_integration_final_pr.h5ad'\n",
    "fn1 = '/burg/tosches/users/mt3353/SAMap/datasets/Pw_neurons_3sp_integration_final_pr.h5ad'\n",
    "fn2 = '/burg/tosches/users/mt3353/SAMap/datasets/Pv_neurons_3sp_integration_final_pr.h5ad'\n",
    "\n",
    "sam1=SAM()\n",
    "sam1.load_data(fn1)\n",
    "\n",
    "sam2=SAM()\n",
    "sam2.load_data(fn2)\n",
    "\n",
    "sam3=SAM()\n",
    "sam3.load_data(fn3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Finally, rather than determining neighborhood sizes by hopping along each cell's outgoing edges, we can also determine them using the cell type labels. In other words, cells that have the same label all belong to the same neighborhood. This can be set using the `neigh_from_keys` parameter of `SAMAP.run`.\n",
    "\n",
    "The SAM files inherited metadata from the original Seurat objects. \"tosches_annot\", \"cluster_label\" and \"unassigned\" are the metadata columns for the cluster names of the lizard, salamander, and turtle data, respectively. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sams = {'pw':sam1,'pv':sam2,'cp':sam3}\n",
    "sm = SAMAP(\n",
    "        sams,\n",
    "        f_maps = '/burg/tosches/users/mt3353/SAMap/processed/',\n",
    "        keys = {'pv':'tosches_annot','pw':'cluster_label', 'cp':'unassigned'},\n",
    "    )\n",
    "sm.run(neigh_from_keys = {'pv':True,'pw':True, 'cp':True})\n",
    "\n",
    "# samap.utils.save_samap(sm, 'pv_pw.sam')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# to save SAMap objects \n",
    "from samap.utils import save_samap\n",
    "\n",
    "save_samap(sm, '/burg/tosches/users/mt3353/SAMap/datasets/3sp_integration_granular_220613_SAM.pkl')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<samap.mapping.SAMAP at 0x1554c9205f40>"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# to import objects that were already saved\n",
    "from samap.utils import load_samap\n",
    "\n",
    "fn = '/burg/tosches/users/mt3353/SAMap/datasets/salamander_lizard_integration_220607_SAM.pkl'\n",
    "\n",
    "sm = SAM()\n",
    "load_samap(fn)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Post-SAMap analysis\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To calculate alignment scores between cell types, we can use `get_mapping_scores`. This function will use the combined SAM object produced by SAMap to calculate alignment scores between cell types in the provided cell type annotation columns of `sam.adata.obs`. \n",
    "\n",
    "The resulting tables show the highest-scoring alignment scores for each cell type (`D`) and pairwise mapping scores between cell types (`MappingTable`)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "keys = {'pv':'tosches_annot','pw':'cluster_label','cp':'unassigned'}\n",
    "D,MappingTable = get_mapping_scores(sm,keys,n_top = 10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"2\" halign=\"left\">pw_TEGLU30</th>\n",
       "      <th colspan=\"2\" halign=\"left\">pv_PThE_2</th>\n",
       "      <th colspan=\"2\" halign=\"left\">pw_TEGABA24</th>\n",
       "      <th colspan=\"2\" halign=\"left\">pv_MGE_INs_3</th>\n",
       "      <th colspan=\"2\" halign=\"left\">pv_Chol</th>\n",
       "      <th>...</th>\n",
       "      <th colspan=\"2\" halign=\"left\">pw_TEGABA53</th>\n",
       "      <th colspan=\"2\" halign=\"left\">pw_TEGABA19</th>\n",
       "      <th colspan=\"2\" halign=\"left\">pw_TEGLU36</th>\n",
       "      <th colspan=\"2\" halign=\"left\">pw_TEGABA29</th>\n",
       "      <th colspan=\"2\" halign=\"left\">pw_TEGLU31</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>...</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>pv_PThE_2</td>\n",
       "      <td>0.998491</td>\n",
       "      <td>pw_TEGLU30</td>\n",
       "      <td>0.998491</td>\n",
       "      <td>pv_MGE_INs_3</td>\n",
       "      <td>0.998261</td>\n",
       "      <td>pw_TEGABA24</td>\n",
       "      <td>0.998261</td>\n",
       "      <td>pw_TEGABA32</td>\n",
       "      <td>0.997721</td>\n",
       "      <td>...</td>\n",
       "      <td>pv_OT_1</td>\n",
       "      <td>0.151136</td>\n",
       "      <td>pv_CGE_INs_1</td>\n",
       "      <td>0.142309</td>\n",
       "      <td>pv_PThE_2</td>\n",
       "      <td>0.140198</td>\n",
       "      <td>pv_CGE_INs_1</td>\n",
       "      <td>0.126669</td>\n",
       "      <td>pv_PThE_2</td>\n",
       "      <td>0.063613</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cp_e17</td>\n",
       "      <td>0.153284</td>\n",
       "      <td>pw_TEGLU35</td>\n",
       "      <td>0.755401</td>\n",
       "      <td>cp_i12</td>\n",
       "      <td>0.979571</td>\n",
       "      <td>pw_TEGABA25</td>\n",
       "      <td>0.876744</td>\n",
       "      <td>pw_TEGABA37</td>\n",
       "      <td>0.287141</td>\n",
       "      <td>...</td>\n",
       "      <td>pv_CGE_INs_1</td>\n",
       "      <td>0.150791</td>\n",
       "      <td>pv_34</td>\n",
       "      <td>0.109761</td>\n",
       "      <td>cp_e02</td>\n",
       "      <td>0.0518</td>\n",
       "      <td>cp_i15</td>\n",
       "      <td>0.072909</td>\n",
       "      <td>cp_e17</td>\n",
       "      <td>0.018194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cp_e10</td>\n",
       "      <td>0.05515</td>\n",
       "      <td>pw_TEGLU39</td>\n",
       "      <td>0.656101</td>\n",
       "      <td>pv_MGE_INs_1</td>\n",
       "      <td>0.886297</td>\n",
       "      <td>cp_i09</td>\n",
       "      <td>0.866054</td>\n",
       "      <td>cp_e17</td>\n",
       "      <td>0.128316</td>\n",
       "      <td>...</td>\n",
       "      <td>cp_e17</td>\n",
       "      <td>0.149877</td>\n",
       "      <td>pv_OT_1</td>\n",
       "      <td>0.08901</td>\n",
       "      <td>cp_e17</td>\n",
       "      <td>0.049424</td>\n",
       "      <td>cp_i16</td>\n",
       "      <td>0.031955</td>\n",
       "      <td>cp_e02</td>\n",
       "      <td>0.01402</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cp_e02</td>\n",
       "      <td>0.028806</td>\n",
       "      <td>cp_e17</td>\n",
       "      <td>0.613492</td>\n",
       "      <td>pv_MGE_INs_2</td>\n",
       "      <td>0.777871</td>\n",
       "      <td>cp_i08</td>\n",
       "      <td>0.814856</td>\n",
       "      <td>pw_TEGABA36</td>\n",
       "      <td>0.075507</td>\n",
       "      <td>...</td>\n",
       "      <td>pv_MGE_INs_3</td>\n",
       "      <td>0.104033</td>\n",
       "      <td>cp_e17</td>\n",
       "      <td>0.072664</td>\n",
       "      <td>cp_e19</td>\n",
       "      <td>0.023714</td>\n",
       "      <td>pv_MGE_INs_3</td>\n",
       "      <td>0.028531</td>\n",
       "      <td>cp_e10</td>\n",
       "      <td>0.008528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cp_e11</td>\n",
       "      <td>0.018458</td>\n",
       "      <td>pw_TEGLU41</td>\n",
       "      <td>0.525004</td>\n",
       "      <td>cp_i09</td>\n",
       "      <td>0.750855</td>\n",
       "      <td>cp_i10</td>\n",
       "      <td>0.652454</td>\n",
       "      <td>cp_e13</td>\n",
       "      <td>0.061634</td>\n",
       "      <td>...</td>\n",
       "      <td>pv_PThE_2</td>\n",
       "      <td>0.077746</td>\n",
       "      <td>pv_MGE_INs_1</td>\n",
       "      <td>0.056158</td>\n",
       "      <td>cp_e11</td>\n",
       "      <td>0.016587</td>\n",
       "      <td>pv_Str_PENK_2</td>\n",
       "      <td>0.022047</td>\n",
       "      <td>pv_Sept_2</td>\n",
       "      <td>0.002218</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 350 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  pw_TEGLU30                   pv_PThE_2                   pw_TEGABA24  \\\n",
       "     Cluster Alignment score     Cluster Alignment score       Cluster   \n",
       "0  pv_PThE_2        0.998491  pw_TEGLU30        0.998491  pv_MGE_INs_3   \n",
       "1     cp_e17        0.153284  pw_TEGLU35        0.755401        cp_i12   \n",
       "2     cp_e10         0.05515  pw_TEGLU39        0.656101  pv_MGE_INs_1   \n",
       "3     cp_e02        0.028806      cp_e17        0.613492  pv_MGE_INs_2   \n",
       "4     cp_e11        0.018458  pw_TEGLU41        0.525004        cp_i09   \n",
       "\n",
       "                  pv_MGE_INs_3                      pv_Chol                  \\\n",
       "  Alignment score      Cluster Alignment score      Cluster Alignment score   \n",
       "0        0.998261  pw_TEGABA24        0.998261  pw_TEGABA32        0.997721   \n",
       "1        0.979571  pw_TEGABA25        0.876744  pw_TEGABA37        0.287141   \n",
       "2        0.886297       cp_i09        0.866054       cp_e17        0.128316   \n",
       "3        0.777871       cp_i08        0.814856  pw_TEGABA36        0.075507   \n",
       "4        0.750855       cp_i10        0.652454       cp_e13        0.061634   \n",
       "\n",
       "   ...   pw_TEGABA53                   pw_TEGABA19                 pw_TEGLU36  \\\n",
       "   ...       Cluster Alignment score       Cluster Alignment score    Cluster   \n",
       "0  ...       pv_OT_1        0.151136  pv_CGE_INs_1        0.142309  pv_PThE_2   \n",
       "1  ...  pv_CGE_INs_1        0.150791         pv_34        0.109761     cp_e02   \n",
       "2  ...        cp_e17        0.149877       pv_OT_1         0.08901     cp_e17   \n",
       "3  ...  pv_MGE_INs_3        0.104033        cp_e17        0.072664     cp_e19   \n",
       "4  ...     pv_PThE_2        0.077746  pv_MGE_INs_1        0.056158     cp_e11   \n",
       "\n",
       "                     pw_TEGABA29                 pw_TEGLU31                  \n",
       "  Alignment score        Cluster Alignment score    Cluster Alignment score  \n",
       "0        0.140198   pv_CGE_INs_1        0.126669  pv_PThE_2        0.063613  \n",
       "1          0.0518         cp_i15        0.072909     cp_e17        0.018194  \n",
       "2        0.049424         cp_i16        0.031955     cp_e02         0.01402  \n",
       "3        0.023714   pv_MGE_INs_3        0.028531     cp_e10        0.008528  \n",
       "4        0.016587  pv_Str_PENK_2        0.022047  pv_Sept_2        0.002218  \n",
       "\n",
       "[5 rows x 350 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "D.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pw_DIMEGABA1</th>\n",
       "      <th>pw_DIMEGABA2</th>\n",
       "      <th>pw_DIMEGLU1</th>\n",
       "      <th>pw_DIMEGLU2</th>\n",
       "      <th>pw_DIMEGLU3</th>\n",
       "      <th>pw_DIMEGLU4</th>\n",
       "      <th>pw_DIMEGLU5</th>\n",
       "      <th>pw_DIMEGLU6</th>\n",
       "      <th>pw_TEGABA1</th>\n",
       "      <th>pw_TEGABA10</th>\n",
       "      <th>...</th>\n",
       "      <th>pv_Sept_1</th>\n",
       "      <th>pv_Sept_2</th>\n",
       "      <th>pv_Str_PENK_1</th>\n",
       "      <th>pv_Str_PENK_2</th>\n",
       "      <th>pv_Str_TAC1</th>\n",
       "      <th>pv_aDVR_1</th>\n",
       "      <th>pv_aDVR_2</th>\n",
       "      <th>pv_aDVR_3</th>\n",
       "      <th>pv_aDVR_4</th>\n",
       "      <th>pv_pDVR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>pw_DIMEGABA1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.008628</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.054077</td>\n",
       "      <td>0.012336</td>\n",
       "      <td>0.001682</td>\n",
       "      <td>0.000903</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.122586</td>\n",
       "      <td>0.009047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pw_DIMEGABA2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.046626</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001794</td>\n",
       "      <td>0.081573</td>\n",
       "      <td>0.006341</td>\n",
       "      <td>0.004030</td>\n",
       "      <td>0.002638</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.067009</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pw_DIMEGLU1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.036991</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pw_DIMEGLU2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.006736</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.003224</td>\n",
       "      <td>0.061044</td>\n",
       "      <td>0.046855</td>\n",
       "      <td>0.231890</td>\n",
       "      <td>0.002063</td>\n",
       "      <td>0.005057</td>\n",
       "      <td>0.207696</td>\n",
       "      <td>0.038520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pw_DIMEGLU3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.013825</td>\n",
       "      <td>0.023879</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 154 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              pw_DIMEGABA1  pw_DIMEGABA2  pw_DIMEGLU1  pw_DIMEGLU2  \\\n",
       "pw_DIMEGABA1           0.0           0.0          0.0          0.0   \n",
       "pw_DIMEGABA2           0.0           0.0          0.0          0.0   \n",
       "pw_DIMEGLU1            0.0           0.0          0.0          0.0   \n",
       "pw_DIMEGLU2            0.0           0.0          0.0          0.0   \n",
       "pw_DIMEGLU3            0.0           0.0          0.0          0.0   \n",
       "\n",
       "              pw_DIMEGLU3  pw_DIMEGLU4  pw_DIMEGLU5  pw_DIMEGLU6  pw_TEGABA1  \\\n",
       "pw_DIMEGABA1          0.0          0.0          0.0          0.0         0.0   \n",
       "pw_DIMEGABA2          0.0          0.0          0.0          0.0         0.0   \n",
       "pw_DIMEGLU1           0.0          0.0          0.0          0.0         0.0   \n",
       "pw_DIMEGLU2           0.0          0.0          0.0          0.0         0.0   \n",
       "pw_DIMEGLU3           0.0          0.0          0.0          0.0         0.0   \n",
       "\n",
       "              pw_TEGABA10  ...  pv_Sept_1  pv_Sept_2  pv_Str_PENK_1  \\\n",
       "pw_DIMEGABA1          0.0  ...   0.008628        0.0       0.054077   \n",
       "pw_DIMEGABA2          0.0  ...   0.046626        0.0       0.001794   \n",
       "pw_DIMEGLU1           0.0  ...   0.000000        0.0       0.000000   \n",
       "pw_DIMEGLU2           0.0  ...   0.006736        0.0       0.003224   \n",
       "pw_DIMEGLU3           0.0  ...   0.000000        0.0       0.000000   \n",
       "\n",
       "              pv_Str_PENK_2  pv_Str_TAC1  pv_aDVR_1  pv_aDVR_2  pv_aDVR_3  \\\n",
       "pw_DIMEGABA1       0.012336     0.001682   0.000903   0.000000   0.000000   \n",
       "pw_DIMEGABA2       0.081573     0.006341   0.004030   0.002638   0.000000   \n",
       "pw_DIMEGLU1        0.000000     0.000000   0.000000   0.000000   0.000000   \n",
       "pw_DIMEGLU2        0.061044     0.046855   0.231890   0.002063   0.005057   \n",
       "pw_DIMEGLU3        0.013825     0.023879   0.000000   0.000000   0.000000   \n",
       "\n",
       "              pv_aDVR_4   pv_pDVR  \n",
       "pw_DIMEGABA1   0.122586  0.009047  \n",
       "pw_DIMEGABA2   0.067009  0.000000  \n",
       "pw_DIMEGLU1    0.000000  0.036991  \n",
       "pw_DIMEGLU2    0.207696  0.038520  \n",
       "pw_DIMEGLU3    0.000000  0.000000  \n",
       "\n",
       "[5 rows x 154 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MappingTable.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save mapping table\n",
    "keys = {'pv':'tosches_annot','pw':'cluster_label','cp':'superclusters'}\n",
    "D,MappingTable = get_mapping_scores(sm,keys,n_top = 0)\n",
    "#cropping mapping table because it contains everything twice:\n",
    "mt = D\n",
    "mt.to_csv('/burg/tosches/users/mt3353/SAMap/datasets/pw_pv_cp_220613_SAM_table.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# saving UMAP coordinates computed by SAMap\n",
    "sams = sm.sams\n",
    "pv_umap_coord = sams['pv'].adata.obsm['X_umap_samap']\n",
    "pw_umap_coord = sams['pw'].adata.obsm['X_umap_samap']\n",
    "cp_umap_coord = sams['cp'].adata.obsm['X_umap_samap']\n",
    "pv_cell_names = sams['pv'].adata.obs_names\n",
    "pw_cell_names = sams['pw'].adata.obs_names\n",
    "cp_cell_names = sams['cp'].adata.obs_names\n",
    "pv_umap = pd.DataFrame(pv_umap_coord, index=pv_cell_names, columns=['UMAP_1','UMAP_2'])\n",
    "pw_umap = pd.DataFrame(pw_umap_coord, index=pw_cell_names, columns=['UMAP_1','UMAP_2'])\n",
    "cp_umap = pd.DataFrame(cp_umap_coord, index=cp_cell_names, columns=['UMAP_1','UMAP_2'])\n",
    "\n",
    "pv_umap.to_csv('/burg/tosches/users/mt3353/SAMap/datasets/3sp_SAMap_UMAP_pv.csv')\n",
    "pw_umap.to_csv('/burg/tosches/users/mt3353/SAMap/datasets/3sp_SAMap_UMAP_pw.csv')\n",
    "cp_umap.to_csv('/burg/tosches/users/mt3353/SAMap/datasets/3sp_SAMap_UMAP_cp.csv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
